haystack/super_components/converters/multi_file_converter.py

```diff
@@ -0,0 +1,162 @@
+                "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet": ".xlsx",
+                "application/vnd.openxmlformats-officedocument.presentationml.presentation": ".pptx",
+            },
+        )
+
+        csv = CSVToDocument(encoding=self.encoding)
+        docx = DOCXToDocument()
+        html = HTMLToDocument()
+        json = JSONConverter(content_key=self.json_content_key)
+        md = MarkdownToDocument()
+        txt = TextFileToDocument(encoding=self.encoding)
+        pdf = PyPDFToDocument()
+        pptx = PPTXToDocument()
+        xlsx = XLSXToDocument()
+
+        joiner = DocumentJoiner()
+
+        # Create pipeline and add components
+        pp = Pipeline()
+
+        pp.add_component("router", router)
+
+        pp.add_component("docx", docx)
+        pp.add_component("html", html)
+        pp.add_component("json", json)
+        pp.add_component("md", md)
+        pp.add_component("txt", txt)
+        pp.add_component("pdf", pdf)
+        pp.add_component("pptx", pptx)
+        pp.add_component("xlsx", xlsx)
+        pp.add_component("joiner", joiner)
+        pp.add_component("csv", csv)
+
+        pp.connect(f"router.{ConverterMimeType.CSV.value}", "csv")
+        pp.connect(f"router.{ConverterMimeType.DOCX.value}", "docx")
+        pp.connect(f"router.{ConverterMimeType.HTML.value}", "html")
+        pp.connect(f"router.{ConverterMimeType.JSON.value}", "json")
+        pp.connect(f"router.{ConverterMimeType.MD.value}", "md")
+        pp.connect(f"router.{ConverterMimeType.TEXT.value}", "txt")
+        pp.connect(f"router.{ConverterMimeType.PDF.value}", "pdf")
+        pp.connect(f"router.{ConverterMimeType.PPTX.value}", "pptx")
+        pp.connect(f"router.{ConverterMimeType.XLSX.value}", "xlsx")
+
+        pp.connect("docx.documents", "joiner.documents")
+        pp.connect("html.documents", "joiner.documents")
+        pp.connect("json.documents", "joiner.documents")
+        pp.connect("md.documents", "joiner.documents")
+        pp.connect("txt.documents", "joiner.documents")
+        pp.connect("pdf.documents", "joiner.documents")
+        pp.connect("pptx.documents", "joiner.documents")
+
+        pp.connect("csv.documents", "joiner.documents")
+        pp.connect("xlsx.documents", "joiner.documents")
+
+        output_mapping = {"joiner.documents": "documents", "router.unclassified": "unclassified"}
+        input_mapping = {"sources": ["router.sources"], "meta": ["router.meta"]}
+
+        super(MultiFileConverter, self).__init__(
+            pipeline=pp, output_mapping=output_mapping, input_mapping=input_mapping
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this instance to a dictionary.
+        """
+        return default_to_dict(self, encoding=self.encoding, json_content_key=self.json_content_key)
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "MultiFileConverter":
```

Comment:
We should check if we can leave out this method. I think we can because MultiFileConverter is just another component